array-methods.js

```diff
@@ -39,7 +66,22 @@ var sumOfInterests = null;
     round this number to the nearest dollar before moving on.
   )
  */
-var stateSums = null;
+
+var stateSums = {};
+
+dataArray.forEach(
```

Comment:
Instead of using Object.defineProperty, you could merge this line and line 70. 

`const stateSums = dataArray.forEach(function (i) {`

Then use a reduce to end up with an object that will be set to stateSums.